### ISSUE TITLE
[core][release] chaos_dataset_shuffle_push_based_sort_1tb failed with WorkerCrashedError

### ISSUE THREAD

**Issue by rickyyx** [ray-project/ray#28774]
### What happened + What you expected to happen

[chaos_dataset_shuffle_push_based_sort_1tb](https://console.anyscale.com/o/anyscale-internal/projects/prj_FKRmeV5pA6X72aVscFALNC32/clusters/ses_rgW51uGKM59NkPdcJcRsuVMk?command-history-section=command_history) 

```
(raylet, ip=172.31.66.10) [2022-09-22 16:02:34,214 E 138 138] (raylet) node_manager.cc:2966: System memory low at node with IP 172.31.66.10. Used memory (54.90GB) / total capacity (57.60GB) (0.95307) exceeds threshold 0.95, killing latest task with name  and task ID NIL_ID to avoid running out of memory.
(raylet, ip=172.31.66.10) This may indicate a memory leak in a task or actor, or that too many tasks are running in parallel.
(raylet, ip=172.31.66.10) To find the highest memory consumers, use `ray logs raylet.out -ip 172.31.66.10`.
(raylet, ip=172.31.66.10) Consider provisioning more memory on this node or reducing task parallelism by requesting more CPUs per task. To adjust the eviction threshold, set the environment variable `RAY_memory_usage_threshold_fraction` when starting Ray. To disable worker eviction, set the environment variable `RAY_memory_monitor_interval_ms` to zero.

....


ray.exceptions.WorkerCrashedError: The worker died unexpectedly while executing this task. Check python-core-worker-*.log files for more information.
```

### Versions / Dependencies

Last success: [45d7cd2](https://github.com/ray-project/ray/commit/45d7cd2f86d0eccdad632ac04514279f4d126106)
First Failure: [fb7472f](https://github.com/ray-project/ray/commit/fb7472f047d669ec964bcb66f5f718c68609864b)

fb7472f04 Remove RAY_RAYLET_NODE_ID (#28715)
93f911eb6 Add API latency and call counts metrics to dashboard APIs (#28279)
66aae4cb3 [Release Test] Make sure to delete all EBS volumes (#28707)
697df80a0 [Serve] [Docs] Remove incorrect output (#28708)
d8c9aa777 [docs] configurable ecosystem gallery (#28662)
42874e1fc [RLlib] Atari gym environments now require ale-py. (#28703)
b7f0346d5 [AIR] Maintain dtype info in LightGBMPredictor (#28673)
f6ae7ee44 [tune] Test background syncer serialization (#28699)
87f22e106 [ci] Requirements contains duplicate of 'starlette' (#28698)
2e7040ef1 [Tune] [PBT] Maintain consistent `Trial`/`TrialRunner` state when pausing and resuming trial (#28511)
653063555 [ci] Fix mac pipeline (use python 2 in CI scripts) (#28695)
ee2a8da87 [ci] Move to new hierarchical docker structure + pipeline (#28641)
a3c97b44e [Doc] Revamp ray core design patterns doc [8/n]: pass large arg by value (#28660)
db2ce69e4 [Datasets] Add initial aggregate benchmark (#28486)
9c2abf916 [KubeRay][Operator] Improve migration notes (#28672)
45d7cd2f8 [core] Support generators to allow tasks to return a dynamic number of objects (#28291)

### Reproduction script

NA

**Comment by rickyyx** [comment]
There are a bunch of open issues but seems to me the root causes might be different: 
- 28242: seems to be different issue
- 28411: seems to be the same issue, but there was a successful run recently, and started to fail again. So this might be different.

**Comment by jjyao** [comment]
The last 10 runs all succeeded without this error.

**Comment by hora-anyscale** [comment]
Per Triage Sync: Closing, test flaky
